content/docs/setup/listeners/tcp.md

```diff
@@ -0,0 +1,203 @@
+   ---
+   apiVersion: v1
+   kind: Service
+   metadata:
+     labels:
+       app: tcp-echo
+     name: tcp-echo
+     namespace: default
+   spec:
+     ports:
+     - name: http
+       port: 1025
+       protocol: TCP
+       targetPort: 1025
+     selector:
+       app: tcp-echo
+   EOF
+   ```
+
+## Set up the Gateway for TCP routes {#tcp-setup}
+
+Create a TCP listener so that the gateway can route TCP traffic. In the following example, all TCP streams on port 8000 of the gateway are forwarded to port 1025 of the example TCP echo service.
+
+1. Create a Gateway resource with a TCP listener. 
+   
+   ```yaml
+   kubectl apply -f- <<EOF
+   apiVersion: gateway.networking.k8s.io/v1
+   kind: Gateway
+   metadata:
+     name: tcp-gateway
+     namespace: gloo-system
+     labels:
+       app: tcp-echo
+   spec:
+     gatewayClassName: gloo-gateway
+     listeners:
+     - protocol: TCP
+       port: 8000
+       name: tcp
+       allowedRoutes:
+         kinds:
+         - kind: TCPRoute
+   EOF
+   ```
+
+   |Setting|Description|
+   |--|--|
+   |`spec.gatewayClassName`|The name of the Kubernetes gateway class that you want to use to configure the gateway. When you set up {{% reuse "docs/snippets/product-name.md" %}}, a default gateway class is set up for you. To view the gateway class configuration, see [Gateway classes and types](docs/about/class-type/). |
+   |`spec.listeners`|Configure the listeners for this gateway. In this example, you configure a TCP gateway that listens for incoming traffic on port 8000. The gateway can serve TCPRoutes from any namespace. |
+
+2. Check the status of the gateway to make sure that your configuration is accepted and no conflicts exist in your cluster. 
+   
+   ```sh
+   kubectl get gateway tcp-gateway -n gloo-system -o yaml
+   ```
+
+   Example output:
+
+   ```console
+   status:
+     addresses:
+     - type: IPAddress
+       value: ${INGRESS_GW_ADDRESS}
+     conditions:
+     - lastTransitionTime: "2024-11-20T16:01:25Z"
+       message: ""
+       observedGeneration: 2
+       reason: Accepted
+       status: "True"
+       type: Accepted
+     - lastTransitionTime: "2024-11-20T16:01:25Z"
+       message: ""
+       observedGeneration: 2
+       reason: Programmed
+       status: "True"
+       type: Programmed
+   ```
+
+3. Create a TCPRoute resource for the TCP echo app that is served by the gateway that you created.
+   
+   ```yaml
+   kubectl apply -f- <<EOF
+   apiVersion: gateway.networking.k8s.io/v1alpha2
+   kind: TCPRoute
+   metadata:
+     name: tcp-route-echo
+     namespace: gloo-system
+     labels:
+       app: tcp-echo
+   spec:
+     parentRefs:
+       - name: tcp-gateway
+         namespace: gloo-system
+         sectionName: tcp
+     rules:
+       - backendRefs:
+           - name: tcp-echo
+             port: 1025
+   EOF
+   ```
+
+4. Verify that the TCPRoute is applied successfully. 
+   
+   ```sh
+   kubectl get tcproute/tcp-route-echo -n gloo-system -o yaml
+   ```
```

Comment:
IF https://github.com/k8sgateway/k8sgateway/pull/10368 is merged,  the Accepted and ResolvedRefs conditions should be True while the expected Gateway should be reflected in the parentRef. For example:
```
status:
  parents:
  - conditions:
    - lastTransitionTime: "2024-11-21T16:22:52Z"
      message: ""
      observedGeneration: 1
      reason: Accepted
      status: "True"
      type: Accepted
    - lastTransitionTime: "2024-11-21T16:22:52Z"
      message: ""
      observedGeneration: 1
      reason: ResolvedRefs
      status: "True"
      type: ResolvedRefs
    controllerName: solo.io/gloo-gateway
    parentRef:
      group: gateway.networking.k8s.io
      kind: Gateway
      name: tcp-gateway
      sectionName: tcp
```